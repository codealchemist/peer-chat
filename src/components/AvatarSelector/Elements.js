import styled from 'styled-components'
import Icon from '@material-ui/core/Icon'

export const styles = theme => ({
  paper: {
    position: 'absolute',
    width: theme.spacing.unit * 50,
    maxWidth: '95%',
    color: 'black',
    boxShadow: theme.shadows[5],
    outline: 'none',
    top: '20%',
    left: 0,
    right: 0,
    margin: 'auto',
    textAlign: 'center',
    height: 'calc(80% - 10px)',
    overflow: 'scroll'
  }
})

export const Wrapper = styled.div`
  width: 56px;
  height: 56px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
`

export const ModalContent = styled.div`
  background: rgba(255, 255, 255, 0.8);
  padding: 10px;
`

export const Title = styled.h4`
  margin-top: 0;
`

export const IconsList = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`

export const StyledIcon = styled(Icon)`
  cursor: pointer;
  :hover {
    color: white;
  }
`
