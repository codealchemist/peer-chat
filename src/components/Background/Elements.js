import styled from 'styled-components'
import getImage from './getImage'

export const Wrapper = styled.div`
  background: url(${getImage()}) no-repeat;
  background-size: cover;
  background-position: center;
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: -1;
  filter: blur(5px);
`

export const Fade = styled.div`
  width: 100%;
  height: 100%;
  background: black;
  opacity: 0.7;
`
