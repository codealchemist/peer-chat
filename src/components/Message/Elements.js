import styled from 'styled-components'

const initiatorBackgroundColor = 'rgba(150, 150, 150, 0.2)'
const peerBackgroundColor = 'rgba(150, 150, 255, 0.2)'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: row-reverse;
  margin-bottom: 5px;

  ${({ type }) => {
    if (type === 'peer') {
      return `
        flex-direction: row;
      `
    }
  }}
`

export const Content = styled.div`
  width: 50%;
  display: flex;
  width: 100%;
  margin: 0 10px;
  justify-content: flex-end;

  ${({ type }) => {
    if (type === 'peer') {
      return `
        justify-content: flex-start;
      `
    }

    if (type === 'notification') {
      return `
        justify-content: center;
      `
    }
  }}
`

export const Date = styled.div`
  position: absolute;
  top: 5px;
  left: 10px;

  p {
    color: #aaa;
    font-size: 0.75rem;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    font-weight: 400;
    letter-spacing: 0.03333em;
    margin: 0;
  }
`

export const Text = styled.div`
  width: 50%;
  background: ${initiatorBackgroundColor};
  color: #eee;
  padding: 20px 10px 5px 10px;
  border-radius: 5px;
  min-width: 220px;
  position: relative;
  word-break: break-word;

  > p {
    word-break: break-word;
  }

  :before {
    content: '';
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 5px 0 5px 7px;
    border-color: transparent transparent transparent
      ${initiatorBackgroundColor};
    position: absolute;
    top: 9px;
    right: -7px;

    ${({ type }) => {
    if (type === 'peer') {
      return `
          border-width: 5px 7px 5px 0;
          border-color: transparent ${peerBackgroundColor} transparent transparent;
          left: -7px;
        `
    }

    if (type === 'notification') {
      return `
        display: none;
      `
    }
  }}
  }

  ${({ type }) => {
    if (type === 'peer') {
      return `
        background: ${peerBackgroundColor};
      `
    }

    if (type === 'notification') {
      return `
        background: rgba(0, 0, 0, 0.2);
      `
    }
  }}
`
