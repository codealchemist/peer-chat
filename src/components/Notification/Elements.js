import styled from 'styled-components'

const backgroundColor = 'rgba(255, 255, 255, 0.5)'

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: 5px;
`

export const Content = styled.div`
  width: 50%;
  display: flex;
  width: 100%;
  margin: 0 10px;
  justify-content: center;
`

export const Date = styled.div`
  position: absolute;
  top: 5px;
  left: 10px;

  p {
    color: #444;
    font-size: 0.75rem;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    font-weight: 400;
    letter-spacing: 0.03333em;
    margin: 0;
  }
`

export const Text = styled.div`
  width: 50%;
  background: ${backgroundColor};
  padding: 20px 10px 5px 10px;
  border-radius: 2px;
  min-width: 220px;
  position: relative;
  display: flex;
  justify-content: space-between;

  > p {
    color: #222;
    font-size: 0.85rem;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    font-weight: 400;
    margin: 0;
  }

  :before {
    content: '';
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 12px 0 12px 15px;
    border-color: transparent transparent transparent ${backgroundColor};
    position: absolute;
    top: 0;
    right: -15px;
    bottom: 0;
    margin: auto;
  }

  :after {
    content: '';
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 12px 15px 12px 0;
    border-color: transparent ${backgroundColor} transparent transparent;
    position: absolute;
    top: 0;
    left: -15px;
    bottom: 0;
    margin: auto;
  }
`

export const ActionButtonWrapper = styled.div`
  position: absolute;
  top: 2px;
  right: 0px;
`
