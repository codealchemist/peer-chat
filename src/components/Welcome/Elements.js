import styled from 'styled-components'
import Card from '@material-ui/core/Card'

export const StyledCard = styled(Card)`
  h2 {
    margin: 0 0 10px;
  }

  p {
    margin: 30px 10px 10px;
  }
`

export const Form = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;

  @media (max-width: 1024px) {
    flex-direction: column;
  }
`
