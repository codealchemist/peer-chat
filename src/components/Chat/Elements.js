import React from 'react'
import styled from 'styled-components'
import Icon from '@material-ui/core/Icon'
import IconButton from '@material-ui/core/IconButton'
import Notification from 'components/Notification'
import TextField from '@material-ui/core/TextField'
import Clipboard from 'react-clipboard.js'

export const Wrapper = styled.div`
  max-width: 640px;
  width: 100%;
  position: absolute;
  height: calc(100% - 20px);
  top: 0;

  @media (max-width: 800px) {
    max-width: 100%;
  }
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  background: rgba(0, 0, 0, 0.5);
  padding: 20px 20px 0 20px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
`

export const Footer = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
`

export const Messages = styled.div`
  flex: 1;
  overflow: scroll;
`

export const StyledTextField = styled(TextField)`
  flex: 1;
`

export const StyledIconButton = styled(IconButton)`
  position: absolute;
  top: -1px;
  right: 10px;
`

export const InitiatorInitNotification = ({ user, shareUrl }) => {
  const ActionButton = (
    <Clipboard data-clipboard-text={shareUrl} component="a">
      <StyledIconButton id="share-link">
        <Icon style={{ fontSize: '16px' }}>insert_link</Icon>
      </StyledIconButton>
    </Clipboard>
  )

  return (
    <Notification
      text={`${user.name} created chat.`}
      date={new Date()}
      actionButton={ActionButton}
    />
  )
}

export const PeerInitNotification = user => (
  <Notification text={`${user.name} joined chat.`} date={new Date()} />
)
