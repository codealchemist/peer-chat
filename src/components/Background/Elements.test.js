import React from 'react'
import { mount } from 'enzyme'
import { Wrapper } from './Elements'

jest.mock('./getImage', () => () => ({
  default: 'rock-background.jpg'
}))

describe('Background / Elements', () => {
  it('should return wrapper with background image', () => {
    const wrapper = mount(<Wrapper />)
    expect(wrapper).toMatchSnapshot()
  })
})
