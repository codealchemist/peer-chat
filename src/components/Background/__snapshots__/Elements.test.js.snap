// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background / Elements should return wrapper with background image 1`] = `
<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "cZuEeF",
          "rules": Array [
            "
  background: url(",
            "default: rock-background.jpg;",
            ") no-repeat;
  background-size: cover;
  background-position: center;
  width: 100%;
  height: 100%;
  position: absolute;
  z-index: -1;
  filter: blur(5px);
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-bdVaJa cZuEeF"
    />
  </StyledComponent>
</styled.div>
`;
